### PR TITLE
Added .parent helper for items and itemGroups

### PR DESCRIPTION
**Pull request by kunagpal**

### CODE REVIEW

**Comment by kunagpal** [comment]
@czardoz Added

File: test/unit/item-group.test.js [hunk 1]
```diff
@@ -14,4 +14,19 @@ describe('ItemGroup', function () {
         });
         expect(groups.length).to.be(4);
     });
+
+    describe('.parentOf checks', function () {
+        it('should work correctly for nested collections', function () {
+            var collection = new Collection(fixtures.nestedCollectionV2);
```
Comment (czardoz) [inline test/unit/item-group.test.js]: Can we not use the fixtures? They're wreaking havoc and we are supposed to get rid of them soon.. every time we add fixtures, someone or the other ends up adding new tests based on them, and then our tests get coupled.

File: test/unit/item-group.test.js [hunk 2]
```diff
@@ -14,4 +14,19 @@ describe('ItemGroup', function () {
...
+
+            collection.forEachItemGroup(function (group) {
+                var parent = group.parentOf();
+
+                while (parent) {
+                    expect(parent).to.have.keys(['description', 'id', 'name', 'items', 'events']);
+                    parent = parent.parentOf();
```
Comment (czardoz) [inline test/unit/item-group.test.js]: Add `parent.id`s to an array, and verify that the order is as expected.

File: test/unit/item.test.js [hunk 3]
```diff
@@ -20,4 +21,21 @@ describe('Item', function () {
...
+
+    describe('.parentOf', function () {
+        it('must return a falsy result for a standalone item', function () {
+            expect(item.parentOf()).to.not.be.ok();
+        });
+
+        it('must correctly return the parent for a provided item', function () {
+            var collection = new Sdk.Collection(fixtures.collectionV2);
+
+            _.forEach(collection.items.members, function (item) {
```
Comment (czardoz) [inline test/unit/item.test.js]: Use `collection.forEachItem()`

Comment (kunagpal) [inline test/unit/item.test.js]: Done.

File: test/unit/item-group.test.js (hunk 1 above)
Comment (kunagpal) [inline test/unit/item-group.test.js]: Okay, I'll add a todo here.

File: test/unit/item.test.js [hunk 4]
```diff
@@ -20,4 +20,23 @@ describe('Item', function () {
             expect(jsonified).to.have.property('event');
         });
     });
+
+    describe('.parent', function () {
+        it('must return a falsy result for a standalone item', function () {
+            expect(item.parent()).to.not.be.ok();
+        });
+
+        it('must correctly return the parent for a provided item', function () {
```
Comment (czardoz) [inline test/unit/item.test.js]: 4 cases:
```
1. nested item (compare with hardcoded expected value)
2. nested itemgroup (compare with hardcoded expected value)
3. first level item (should return collection as parent)
4. first level itemgroup (should return collection as parent)

Comment (kunagpal) [inline test/unit/item.test.js]: Added

**Pull request merged**